### PR TITLE
Fix Storage Class error handling

### PR DESCRIPTION
**Pull request by alecrajeev**
This pull request will add code to check for a valid storage class.

This is in reference to Issue #1652 [https://github.com/spulec/moto/issues/1652](https://github.com/spulec/moto/issues/1652)

### CODE REVIEW

**Comment by spulec** [comment]
It looks like Travis isn't happy with some of the changes.

**Comment by terrycain** [comment]
If not solved by then, I'll probably attempt to debug this tomorrow evening.

**Comment by alecrajeev** [comment]
I think it's better now, but I am open to potential changes

File: moto/s3/models.py [hunk 1]
```diff
@@ -676,6 +678,8 @@ def get_bucket_website_configuration(self, bucket_name):
 
     def set_key(self, bucket_name, key_name, value, storage=None, etag=None):
         key_name = clean_key_name(key_name)
+        if storage is not None and not(storage in ["STANDARD", "REDUCED_REDUNDANCY", "STANDARD_IA", "ONEZONE_IA"]):
```
Comment (terrycain) [inline moto/s3/models.py]: I think
```python
if storage is not None and storage not in ("STANDARD", "REDUCED_REDUNDANCY", "STANDARD_IA", "ONEZONE_IA")):
```
Looks better :wink: though functionally equivalent, same for line 71.

Would it not make sense to move the list of storage classes to an allcaps variable at the top of the file, then everywhere can reference one copy

File: tests/test_s3/test_s3_storageclass.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	client = boto3.client("s3")
+	client.create_bucket(Bucket="Bucket")
+
+	# add an object to the bucket with standard storage
+
+	client.put_object(Bucket="Bucket", Key="my_key", Body="my_value")
+
+	D = client.list_objects(Bucket="Bucket")
+
+	(D['Contents'][0]["StorageClass"]).should.equal("STANDARD")
```
Comment (terrycain) [inline tests/test_s3/test_s3_storageclass.py]: Are the brackets necessary?

**Comment by terrycain** [comment]
Made 2 comments on more style based things, not really fussed about those, but another unit test would be appreciated. Other than that, great work man

**Comment by alecrajeev** [comment]
Thanks! I can add those changes.

File: tests/test_s3/test_s3_storageclass.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+from __future__ import unicode_literals
+
+import boto
+import boto3
+from boto.exception import S3CreateError, S3ResponseError,
```
Comment (terrycain) [inline tests/test_s3/test_s3_storageclass.py]: TravisCI is complaining about the trailing comma

**Comment by terrycain** [comment]
Looks great, will merge once travis passes.

File: tests/test_s3/test_s3_storageclass.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	s3 = boto3.client("s3")
+	s3.create_bucket(Bucket="Bucket")
+	s3.put_object(Bucket="Bucket", Key="First_Object", Body="Body", StorageClass="ONEZONE_IA")
+
+	s3.create_bucket(Bucket="Bucket2")
+	s3.put_object(Bucket="Bucket2", Key="Second_Object", Body="Body2", StorageClass="STANDARD")
+
+	s3.copy_object(CopySource = {"Bucket": "Bucket", "Key": "First_Object"}, Bucket="Bucket2", Key="Second_Object")
+
+	list_of_copied_objects = client.list_objects(Bucket="Bucket2")
```
Comment (terrycain) [inline tests/test_s3/test_s3_storageclass.py]: Referencing `client` here, it isnt defined, should be `s3`

**Comment by terrycain** [comment]
Looks good, great job man.

**Pull request merged**